libs/core/src/integration/outbox.schema.ts

```diff
@@ -10,161 +9,42 @@ const BaseOutboxProperties = z.object({
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadBodyUpdated),
-      event_payload: events.DiscordThreadBodyUpdated,
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadTitleUpdated),
-      event_payload: events.DiscordThreadTitleUpdated,
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadDeleted),
-      event_payload: events.DiscordThreadDeleted,
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadCommentCreated),
-      event_payload: events.DiscordThreadCommentCreated,
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadCommentUpdated),
-      event_payload: events.DiscordThreadCommentUpdated,
-    })
-    .merge(BaseOutboxProperties),
-  z
-    .object({
-      event_name: z.literal(EventNames.DiscordThreadCommentDeleted),
-      event_payload: events.DiscordThreadCommentDeleted,
-    })
-    .merge(BaseOutboxProperties),
-]);
+const outboxEvents = {
+  [E.ChainEventCreated]: P.ChainEventCreated,
+  [E.CommentCreated]: P.CommentCreated,
+  [E.CommentUpvoted]: P.CommentUpvoted,
+  [E.CommunityCreated]: P.CommunityCreated,
+  [E.ContestContentAdded]: P.ContestContentAdded,
+  [E.ContestContentUpvoted]: P.ContestContentUpvoted,
+  [E.ContestStarted]: P.ContestStarted,
+  [E.DiscordMessageCreated]: P.DiscordMessageCreated,
+  [E.DiscordThreadBodyUpdated]: P.DiscordThreadBodyUpdated,
+  [E.DiscordThreadCommentCreated]: P.DiscordThreadCommentCreated,
+  [E.DiscordThreadCommentDeleted]: P.DiscordThreadCommentDeleted,
+  [E.DiscordThreadCommentUpdated]: P.DiscordThreadCommentUpdated,
+  [E.DiscordThreadCreated]: P.DiscordThreadCreated,
+  [E.DiscordThreadDeleted]: P.DiscordThreadDeleted,
+  [E.DiscordThreadTitleUpdated]: P.DiscordThreadTitleUpdated,
+  [E.FarcasterCastCreated]: P.FarcasterCastCreated,
+  [E.FarcasterReplyCastCreated]: P.FarcasterReplyCastCreated,
+  [E.FarcasterVoteCreated]: P.FarcasterVoteCreated,
+  [E.GroupCreated]: P.GroupCreated,
+  [E.OneOffContestManagerDeployed]: P.OneOffContestManagerDeployed,
+  [E.RecurringContestManagerDeployed]: P.RecurringContestManagerDeployed,
+  [E.SnapshotProposalCreated]: P.SnapshotProposalCreated,
+  [E.SubscriptionPreferencesUpdated]: P.SubscriptionPreferencesUpdated,
+  [E.ThreadCreated]: P.ThreadCreated,
+  [E.ThreadUpvoted]: P.ThreadUpvoted,
+  [E.UserMentioned]: P.UserMentioned,
+} as const;
+
+export const Outbox = z.union(
+  Object.entries(outboxEvents).map(([event_name, event_payload]) =>
+    z
+      .object({
+        event_name: z.literal(event_name as keyof typeof outboxEvents),
+        event_payload,
+      })
+      .merge(BaseOutboxProperties),
+  ) as unknown as readonly [z.ZodTypeAny, z.ZodTypeAny, ...z.ZodTypeAny[]],
```

Comment:
While this is more concise, it completely removes the types on the Outbox schema. Do we really want to do that?

Reply:
I just saw your comment regarding the size of the type. I guess this is fine as long as we make sure to primarily use the `emitEvent` util.

Reply:
Yeah, I think we only need it to validate. I didn't find anything using these types.